### ISSUE TITLE
Static Analysis reporting

### ISSUE THREAD

**Issue by k-dovgan** [k-dovgan/test-repo#67]
Originally created on Tue, 28 Mar 2017 21:34:04 +0900

Add report of Static Analysis results into Code Review system:
1. On each change collect Static Analysis data (issues) from the latest sources without the change
2. Collext list of issues from Static Analsys from the latest sources with the change
3. Calculate the difference
4. Report each new issue right into Code Review system into corresponding line of code

**Comment by k-dovgan** [comment]
Originally posted by i.keliukh on Tue, 24 Oct 2017 20:22:12 +0900

Discussed details as follows:
* Support of static analysis steps must be added to project configs
* Support of different source code-related metrics/reports must be done by external scripts. Scripts must report results in unified format (see Sonar Qube): file, line, message code, message name, description
* Universum must be able to process reports in unified format

At first stage it is enough to collect reports from one run of analysis and report to corresponding places in the code review system. Differential analysis must be implemented later:
* Analysis must be done before and after unshelving
* The analysis step itself can be included into shelved files as a part of configs.py content
Knowing results of analysis after unshelving, universum must process each report and look for it in the original code
* If it was present in original code at the same place (considering possible line-level offset due to changes, see difflib), it is skipped
* If the report is new, or the line is changed, it is reported

**Comment by k-dovgan** [comment]
Originally posted by k.dovgan on Thu, 25 Jan 2018 16:58:33 +0200

Also non-differential analysis should be implemented.
